src/main/java/org/wise/portal/presentation/web/controllers/archive/ArchiveProjectController.java

```diff
@@ -0,0 +1,118 @@
+package org.wise.portal.presentation.web.controllers.archive;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.json.JSONException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.wise.portal.dao.ObjectNotFoundException;
+import org.wise.portal.domain.project.Project;
+import org.wise.portal.domain.project.impl.ProjectImpl;
+import org.wise.portal.domain.user.User;
+import org.wise.portal.domain.usertag.UserTag;
+import org.wise.portal.presentation.web.controllers.ControllerUtil;
+import org.wise.portal.presentation.web.response.ResponseEntityGenerator;
+import org.wise.portal.service.project.ProjectService;
+import org.wise.portal.service.usertags.UserTagsService;
+
+@RestController
+@Secured({ "ROLE_TEACHER" })
+@RequestMapping("/api")
+public class ArchiveProjectController {
+
+  @Autowired
+  private ProjectService projectService;
+
+  @Autowired
+  private UserTagsService userTagsService;
+
+  private static final String ARCHIVED_TAG = "archived";
+
+  @PutMapping("/project/{projectId}/archived")
+  protected ResponseEntity<Map<String, Object>> archiveProject(
+      @PathVariable("projectId") ProjectImpl project) throws JSONException {
+    User user = ControllerUtil.getSignedInUser();
```

Comment:
We're trying to move away from this static calls because they're harder to test. Use ```userService.retrieveUserByUsername(auth.getName())``` instead. See UserAPIController.getStudentUserInfoById() for reference.